src/amms.py

```diff
@@ -7,297 +7,609 @@
+        raise NotImplementedError("Implement me!")
+
+    @staticmethod
+    def convert_to_big_integer_and_decimals(
+        amount_str: str | None,
+    ) -> Tuple[int | None, int | None]:
+        """
+        Convert string containing numerical value into integer and number of decimals.
+        """
+        if amount_str is None:
+            return None, None
+        # Check if there is a decimal point in the amount_str
+        if "." in amount_str:
+            # Split the string into whole and fractional parts
+            whole_part, fractional_part = amount_str.split(".")
+            # Calculate the number of decimals as the length of the fractional part
+            decimals = len(fractional_part)
+            # Remove the decimal point and convert the remaining string to an integer
+            amount_big_integer = int(whole_part + fractional_part)
+        else:
+            # If there is no decimal point, there are no decimals and the conversion is straightforward
+            decimals = 0
+            amount_big_integer = int(amount_str)
+
+        return amount_big_integer, decimals
 
 
 class OrcaAMM(Amm):
-	DEX_NAME = 'Orca'
-
-	def get_pools(self) -> None:
-		"""
-		Fetches pool data from the Orca API and stores it in the `pools` attribute.
-		"""
-		LOG.info("Fetching pools from Orca API")
-		try:
-			result = requests.get('https://api.mainnet.orca.so/v1/whirlpool/list', timeout=30)
-			decoded_result = result.content.decode('utf-8')
-			self.pools = json.loads(decoded_result)['whirlpools']
-			LOG.info(f"Successfully fetched {len(self.pools)} pools")
-		except requests.exceptions.Timeout:
-			LOG.error("Request to Orca whirlpool API timed out")
-			self.get_pools()
-
-	def store_pool(self, pool: Dict[str, Any]) -> None:
-		"""
-		Save pool data to database.
-		"""
-		pair = f"{pool['tokenA']['symbol']}-{pool['tokenB']['symbol']}"
-		market_address = pool.get('address', 'Unknown')
-
-		with get_db_session() as session:
-			# Creating an instance of AmmLiquidity
-			liquidity_entry = AmmLiquidity(
-				timestamp=self.timestamp,
-				dex=self.DEX_NAME,
-				pair=pair,
-				market_address=market_address,
-				additional_info=json.dumps(pool)
-			)
-			# Add to session and commit
-			session.add(liquidity_entry)
-			session.commit()
+    DEX_NAME = "Orca"
+
+    def get_pools(self) -> None:
+        """
+        Fetches pool data from the Orca API and stores it in the `pools` attribute.
+        """
+        LOG.info("Fetching pools from Orca API")
+        try:
+            result = requests.get(
+                "https://api.mainnet.orca.so/v1/whirlpool/list", timeout=30
+            )
+            decoded_result = result.content.decode("utf-8")
+            self.pools = json.loads(decoded_result)["whirlpools"]
+            LOG.info(f"Successfully fetched {len(self.pools)} pools")
+        except requests.exceptions.Timeout:
+            LOG.error("Request to Orca whirlpool API timed out")
+            self.get_pools()
```

Comment:
Probably we could, but I think it would be better do store a list of pools we want to collect and then fetch onchain info for them, basically skipping the API (since the info we need isn't provided anyways). So this method most likely won't be used in future iterations. 

However this PR was intended for BonkSwap/Dooar only so I feel like it would be better to change in other PR as this one's getting pretty hard to navigate with all the changes.